### PR TITLE
feat(langchainjs): add retrieval and invocation parameters support to tracer

### PR DESCRIPTION
**Pull request by Parker-Stafford**
Adds support for retrieval documents on retriever spans and invocation parameters / models on llm spans

### Retrieval
image

### Attributes
image

### LLM Invocation params
image

### Attributes
image

### CODE REVIEW

File: js/packages/openinference-instrumentation-langchain/src/utils.ts [hunk 1]
```diff
@@ -18,9 +20,14 @@ import {
   LLMMessageFunctionCall,
   LLMMessageToolCalls,
   LLMMessagesAttributes,
+  LLMOpenInferenceAttributes,
+  RetrievalDocument,
   SafeFunction,
 } from "./types";
 
+export const RETRIEVAL_DOCUMENTS =
+  `${SemanticAttributePrefixes.retrieval}.${RetrievalAttributePostfixes.documents}` as const;
```
Comment (mikeldking) [inline js/packages/openinference-instrumentation-langchain/src/utils.ts:29]: not really a util, I'd put it closer to where you are using it.

File: js/packages/openinference-instrumentation-langchain/src/utils.ts [hunk 2]
```diff
@@ -37,6 +44,8 @@ export function withSafety<T extends GenericFunction>(fn: T): SafeFunction<T> {
   };
 }
 
+const safelyStringify = withSafety(JSON.stringify);
```
Comment (mikeldking) [inline js/packages/openinference-instrumentation-langchain/src/utils.ts:47]:
```suggestion
const safelyJSONStringify = withSafety(JSON.stringify);
```

File: js/packages/openinference-instrumentation-langchain/test/langchain.test.ts [hunk 3]
```diff
@@ -9,15 +9,35 @@ import { ChatPromptTemplate } from "@langchain/core/prompts";
...
 import {
   OpenInferenceSpanKind,
+  RetrievalAttributePostfixes,
+  SemanticAttributePrefixes,
   SemanticConventions,
 } from "@arizeai/openinference-semantic-conventions";
 import { LangChainTracer } from "../src/tracer";
 import { trace } from "@opentelemetry/api";
-jest.useFakeTimers();
+// jest.useFakeTimers();
```
Comment (mikeldking) [inline js/packages/openinference-instrumentation-langchain/test/langchain.test.ts:23]: intentional?

**Review by mikeldking — APPROVED** [review]
🚀

File: js/packages/openinference-instrumentation-langchain/test/langchain.test.ts (hunk 3 above)
Comment (Parker-Stafford) [reply js/packages/openinference-instrumentation-langchain/test/langchain.test.ts]: yeah i have to uncomment it to send to phoenix, fixed it for the pr now

File: js/packages/openinference-instrumentation-langchain/src/utils.ts (hunk 1 above)
Comment (Parker-Stafford) [reply js/packages/openinference-instrumentation-langchain/src/utils.ts:29]: I think this should just be added to the semantic conventions package, will open a pr.

**Pull request merged**